### PR TITLE
Migrate Label component to TS

### LINKED ISSUE #19123: Migrate components to TS: Label
### Description
Migrate `Label` component to TypeScript

### Technical Details

- Convert Objects in `SNAKE_CASE` to enums in `PascalCase` (name and keys)
- Ensure enums use singular not plural e.g. `Size` not `Sizes`
- Update file name from `component-name.contants.js` to `component-name.types.ts` and add types
- Update file name from `component-name.js` to `component-name.tsx` and add types
- Update file name from `component-name.test.js` to `component-name.test.tsx`
- Update file name from `component-name.stories.js` to `component-name.stories.tsx`
- Update all instances of component objects to enums
- Update all documentation

### Acceptance Criteria

- Component has been converted to TypeScript
- Stories, documentation and tests have been updated inside of the `component-name` folder
- Before/After screen cast of component stories in storybook to show no visual regression after migration

### Difficulty: Intermediate

Good first issue for: External contributors who are familiar with TypeScript and want to contribute to improving the stability of the extension

### PR DESCRIPTION
**Pull request by gabrielmj23**
## Explanation

Fixes #19123 

Update `Label` component in https://github.com/MetaMask/metamask-extension/tree/e5fdb72acc03ce1932b7ad51c788f141a884acd9/ui/components/component-library/label to Typescript.

I've also gone ahead and changed the `DISPLAY` and `FLEX_DIRECTION` constant helper objects in the design system into enums. I'm seeing there's an open issue for this, but I've kept the same expected enum names.

Also added an optional `htmlFor` field to the `Text` component, given that `Label` uses it to render and Typescript doesn't like passing the `htmlFor` field when it's not part of `Text`. Shouldn't break anything as it's optional, and specified in documentation.

## Screenshots/Screencaps

No visual regression was observed when checking the component on Storybook.
### Before
Screenshot 2023-07-20 at 1 46 30 PM

### After 
Screenshot 2023-07-20 at 1 46 25 PM

## Manual Testing Steps

Running `yarn jest ui/components/component-library/label`, passes all tests for this component.

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [X] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [X] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Comment by gabrielmj23** [comment]
I have read the CLA Document and I hereby sign the CLA

File: ui/components/component-library/label/label.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  TextVariant,
+  Display,
+  AlignItems,
+} from '../../../helpers/constants/design-system';
+
+export const Label: React.FC<{
+  htmlFor?: string;
+  className?: string;
+  children: string | React.ReactNode;
+}> = ({ htmlFor, className, children, ...props }) => (
```
Comment (georgewrmarshall) [inline ui/components/component-library/label/label.tsx:15]: Would it be possible to move these to a `.types.ts` file and export the `LabelProps` that will match our other component conventions. Could we also forwardRef to the Text component. I think all of our components should forward refs

File: ui/components/component-library/label/label.stories.tsx [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import React, { useState } from 'react';
```
Comment (georgewrmarshall) [inline ui/components/component-library/label/label.stories.tsx:1]: Could we also include the storybook types checkout `badge-wapper.stories.tsx`

**Review by georgewrmarshall — CHANGES_REQUESTED** [review]
Hey @gabrielmj23, thanks for your contribution this is looking great. There are some conventions in other TS component files that would be great if we could follow for the `Label` component. You could checkout the `BadgeWrapper` component files. In particular the `label.types.ts` file

File: ui/components/component-library/label/label.tsx (hunk 1 above)
Comment (gabrielmj23) [reply ui/components/component-library/label/label.tsx]: Thanks for the suggestions, I made two commits with the changes

File: ui/components/component-library/text/text.types.ts [hunk 3]
```diff
@@ -108,6 +108,10 @@ export interface TextProps extends BoxProps {
    * Changes the root html element tag of the Text component.
    */
   as?: ValidTag;
+  /**
+   * The id of the input associated with the label (in case Text is used as a Label)
+   */
+  htmlFor?: string;
```
Comment (georgewrmarshall) [inline ui/components/component-library/text/text.types.ts:114]: Going to work on a fixing the Box types to allow for polymorphic props so we can pass html attributes down to the html element without having to adding types to the original types. We can merge this and I'll work on it next.

**Review by georgewrmarshall — DISMISSED** [review]
LGTM! Thanks for your contribution 💯 
- checked storybook

**Comment by georgewrmarshall** [comment]
Oh look like you may have to rebase

**Comment by gabrielmj23** [comment]
Tried to rebase there, hope it works, let me know

**Comment by georgewrmarshall** [comment]
Hey @gabrielmj23, Thanks for your time and effort in creating this pull request. We have identified an issue related to the typing of the `Box` component, specifically with the polymorphic `as` prop. To address this issue, we have an open ticket at https://github.com/MetaMask/metamask-extension/issues/19239 and a draft pull request at https://github.com/MetaMask/metamask-extension/pull/19363 that needs to be merged before we can proceed with your PR. Thanks for your patience

**Comment by georgewrmarshall** [comment]
Blocked by https://github.com/MetaMask/metamask-extension/pull/19572

**Review by georgewrmarshall — APPROVED** [review]
LGTM!

**Review by pedronfigueiredo — APPROVED** [review]

**Pull request merged**